indico/queries/teach_tasks.py

```diff
@@ -0,0 +1,244 @@
+
+    Args:
+        dataset_id (int): The id of the dataset to add labels to.
+        labelset_id (int): The id of the labelset to add labels to.
+        target (List[Any]): A list of labels to add to the labelset.
+        row_index (List[Any]): The row indices corresponding with each item in targets.
+
+    Raises:
+        IndicoInputError if the length of targets and row_index are not equal
+
+    """
+
+    query = """
+        mutation(
+            $labels: [SubmissionLabel]!,
+            $dataset_id: Int!,
+            $labelset_id: Int!,
+        ){
+            submitLabels(
+                datasetId: $dataset_id,
+                labelsetId: $labelset_id,
+                labels: $labels
+            ){ success }
+        }
+        """
+
+    def __init__(
+        self, dataset_id: int, labelset_id: int, target: List[Any], row_index: List[int]
+    ):
+        if len(target) != len(row_index):
+            raise IndicoInputError("Mismatch in lengths between target and row_index.")
+
+        labels = []
+        for t, row_id in zip(target, row_index):
+            if t is not None:
+                labels.append({"rowIndex": row_id, "target": json.dumps(t)})
+        super().__init__(
+            query=self.query,
+            variables={
+                "labels": labels,
+                "dataset_id": dataset_id,
+                "labelset_id": labelset_id,
+            },
+        )
+
+
+class GetQuestionaireExamples(GraphQLRequest):
+    """
+    Gets unlabeled examples from a Questionnaire.
+
+    Args:
+        questionaire_id (int): The id of the questionnaire to get examples from.
+        num_examples (int): The number of examples to get from the questionaire.
+
+    Returns:
+        List[Example]
+
+    """
+
+    query = """
+    query(
+        $questionaire_id: Int!,
+        $num_examples: Int!
+    )
+    {
+        questionnaires(questionnaireIds: [$questionaire_id]) {
+            questionnaires {
+                examples(numExamples: $num_examples) {
+                    rowIndex
+                    datafileId
+                    source
+                }
+            }
+        }
+    }
+    """
+
+    def __init__(self, questionaire_id: int, num_examples: int):
+        super().__init__(
+            query=self.query,
+            variables={
+                "questionaire_id": questionaire_id,
+                "num_examples": num_examples,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            examples = [
+                Example(**e)
+                for e in super().process_response(response)["questionnaires"][
+                    "questionnaires"
+                ][0]["examples"]
+            ]
+        except IndexError:
+            raise IndicoNotFound(
+                "Examples not found. Please check the ID you are using."
+            )
+        return examples
+
+
+class CreateQuestionaire(GraphQLRequest):
```

Comment:
The creation is actually asynchronous and the questionnaire has a state. It's probably better to consider this a chain and have a wait keyword to check